src/stores/FrameStore.ts

```diff
@@ -1113,7 +1113,13 @@ export class FrameStore {
     };
 
     private convertSpectral = (values: Array<number>): Array<number> => {
-        return values && values.length > 0 ? values.map(value => this.astSpectralTransform(this.spectralType, this.spectralUnit, this.spectralSystem, value)) : null;
```

Comment:
`this.astSpectralTransform` checks if `this.spectralFrame` and `value` are valid. Maybe we could add the checks back in `convertSpectral` (at least for `this.spectralFrame`)

Reply:
I've added a check to `this.spectralFrame`. If `values` is undefined, then `values?.length` will also be undefined, so this already checks that `values` is valid.